src/app/core/services/repo-creator.service.ts

```diff
@@ -29,4 +61,25 @@ export class RepoCreatorService {
       })
     );
   }
+
+ /**
+  * If a Session is unavailable (because the repository is missing) attempt to create IF it is
+  * the BugReporting Phase
+  * @param currentPhase the current phase of the session.
+  * @param phaseRepo the name of the specified repository.
+  * @return - Dummy Observable to give the API sometime to propagate the creation of the new repository since
+  *           the API Call used here does not return any response.
+  */
+ private attemptSessionAvailabilityFix(currentPhase: Phase, phaseRepo: string): Observable<any> {
+   if (currentPhase !== Phase.phaseBugReporting) {
+     throw new Error(CURRENT_PHASE_REPO_CLOSED);
+   } else if (currentPhase === Phase.phaseBugReporting && this.userService.currentUser.role !== UserRole.Student) {
```

Comment:
We can simplify the condition, since the first branch verifies `currentPhase` already
```suggestion
   } else if (this.userService.currentUser.role !== UserRole.Student) {
```

Reply:
👍 aights